content/en/blog/_posts/2023-12-13-kubernetes-1.29.md

```diff
@@ -0,0 +1,218 @@
+
+To meet this requirement, the CSI Node Expand Secret feature was introduced in Kubernetes v1.25. This allows an optional secret field to be sent as part of the NodeExpandVolumeRequest by the CSI drivers so that node volume expansion operation can be performed with the underlying storage system. In Kubernetes v1.29, this feature became generally available.
+
+### KMS v2 encryption at rest graduates to GA in v1.29 ([Sig Auth](https://github.com/kubernetes/community/tree/master/sig-auth))
+ 
+One of the first things to consider when securing a Kubernetes cluster is encrypting persisted API data at rest. KMS provides an interface for a provider to utilize a key stored in an external key service to perform this encryption. With the new release of Kubernetes, KMS v2 has become a stable feature bringing numerous improvements in performance, key rotation, health check & status, and observability. These enhancements enable users with a reliable solution to encrypt all resources in their Kubernetes clusters. [KEP-3299](https://kep.k8s.io/3299)
+
+For Kubernetes v1.29, [encryption at rest for API data](/docs/tasks/administer-cluster/encrypt-data/)
+can integrate with an external _key management service_ (KMS). Available since Kubernetes v1.13,
+this mechanism allows you to protect your cluster's control plane from security risks even if the etcd
+persistence layer is somehow compromised.
+
+New in v1.29 is stable support for version 2 of the KMS integration. If you're using KMS for
+encryption, or you plan to, the Kubernetes project recommends using the version 2 integration
+mechanism in all cases.
+
+For details about integration with specific cloud providers and other services, ask your provider
+or consult the documentation for their plugin.
+
+## Improvements that graduated to beta in Kubernetes v1.29 {#graduations-to-beta}
+
+_This is a selection of some of the improvements that are now beta following the v1.29 release._
+
+### Per-plugin callback functions in kube-scheduler for accurate requeueing ([SIG Scheduling](https://github.com/kubernetes/community/tree/master/sig-scheduling)) {#scheduler-queueing-hint}
+
+The throughput of the scheduler is our eternal challenge. This QueueingHint feature brings a new possibility to optimize the efficiency of requeueing, which could reduce useless scheduling retries significantly.
+
+### Node lifecycle separated from taint management ([SIG Scheduling](https://github.com/kubernetes/community/tree/master/sig-scheduling))
+
+As title describes, it's to decouple `TaintManager` that performs taint-based pod eviction from `NodeLifecycleController` and make them two separate controllers: `NodeLifecycleController` to add taints to unhealthy nodes and `TaintManager` to perform pod deletion on nodes tainted with NoExecute effect.
+
+### Reduce secret based SA token, LegacyServiceAccountTokenCleanUp going to Beta in v1.29 ([Sig Auth](https://github.com/kubernetes/community/tree/master/sig-auth))
```

Comment:
```suggestion
### Clean up for legacy Secret-based ServiceAccount tokens ([SIG Auth](https://github.com/kubernetes/community/tree/master/sig-auth)) {#serviceaccount-token-clean-up}
```

Reply:
The text I'm amending is tautological, and I'm worried that it's confusing. I wouldn't accept it for docs; I will use my maintainer access as a blog editor to reword.

If we want the wording reinstated, we can do that.